### DISCUSSION TITLE
Does Rocket/BOOM implement any type of IOMMU?

### DISCUSSION THREAD

**Discussion by sammy17** [chipsalliance/rocket-chip#3180] [Q&A] [answered]
I noticed that IOMMU specification is still a work in progress for RISC-V which provides critical protection against DMA attacks (being able to access privileged/kernel memory through DMA). 

Is there any IOMMU implementation or something equivalent that prevents an HW component (e.g.- an accelerator) that has DMA from accessing all of the physical memory in the latest version of Rocket chip?

In other words, what I am asking is, are RISC-V systems (SoCs) that have DMA, vulnerable to DMA attacks? 

Really appreciate any thoughts on this.

**Comment by jerryz123** [comment, marked as answer]
I don't believe there are any IOMMU implementations for rocket-chip floating around. 

A spec-compliant implementation would be a worthy contribution to the rocket-chip repo, if someone is willing. I'm not very familiar with the IOMMU spec, perhaps a minimal implementation is not too burdensome.

**Reply by sammy17** [reply]
I see, thanks for the information!
